apis/apps/v1alpha1/opsrequest_types.go

```diff
@@ -311,12 +311,50 @@ type Upgrade struct {
 	//
 	// This field is deprecated since v0.9 because ClusterVersion is deprecated.
 	//
-	// +kubebuilder:validation:Required
 	// +kubebuilder:deprecatedversion:warning="This field has been deprecated since 0.9.0"
-	ClusterVersionRef string `json:"clusterVersionRef"`
+	ClusterVersionRef string `json:"clusterVersionRef,omitempty"`
+
+	// Lists components to be upgrade based on desired ComponentDefinition and ServiceVersion.
+	// Support the following user scenarios:
+	// 1. empty componentDefinitionName + non-empty serviceVersion:
+	//    Upgrade to the specified version and latest compatible ComponentDefinition.
+	//    But the prerequisite for this scenario is that cluster.spec.clusterDefinition must not be empty.
```

Comment:
why cluster.spec.clusterDefinition must not be empty, it should be all right as long as cluster.spec.clusterDefinition is empty but cluster.spec.components[..].componentdef is not empty.